contrib/query-loc-0.4.0/query-loc.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH QUERY-LOC 1 "January 11, 2005"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
query-loc \- to retrieve and display the location information in the DNS
.SH SYNOPSIS
.B query-loc
.RI [-v] [-d nnn] " host"
.SH DESCRIPTION
This manual page documents briefly the
.B query-loc
command.
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics, 
.\" respectively.
\fBquery-loc\fP is a program to retrieve and display the location
information in the DNS. 

It uses the algorithms described in
RFC 1876 (and RFC 1101 to get the network names).
You can find examples of networks wchich implement this scheme
in the ADDRESSES file.
 
.SH OPTIONS
.TP
.B \-v
Verbose mode. 
.TP
.B \-d nnn
Debug mode. Displays the RFC's algorithm

.SH BUGS

Very few hosts have location information.

.SH AUTHOR
This manual page was written by Stephane Bortzmeyer
<bortzmeyer@debian.org>.

.\" $Id: query-loc.1,v 1.1.4.1 2008-02-15 02:19:42 marka Exp $


CHANGES

2640.	[security]	A specially crafted update packet will cause named
			to exit. [RT #20000]

2375.   [security]      Fully randomize UDP query ports to improve
			forgery resilience. [RT #17949]

	--- 9.3.4-P1 released ---

2203.	[security]	Query id generation was cryptographically weak.
			[RT # 16915]

2193.	[port]		win32: BINDInstall.exe is now linked statically.
			[RT #16906]

2192.	[port]		win32: use vcredist_x86.exe to install Visual
			Studio's redistributable dlls if building with
			Visual Stdio 2005 or later.

	--- 9.3.4 released ---

2126.	[security]	Serialise validation of type ANY responses. [RT #16555]

2124.	[security]	It was possible to dereference a freed fetch
			context. [RT #16584]

	--- 9.3.3 released ---

2107.	[bug]		dighost.c: more cleanup of buffers. [RT #16499]

2104.	[port]		Fix Solaris SMF error message.

2103.	[port]		Add /usr/sfw to list of locations for OpenSSL
			under Solaris.

2102.	[port]		Silence solaris 10 warnings.

2101.	[bug]		OpenSSL version checks were not quite right.
			[RT #16476]

2100.	[port]		win32: copy libeay32.dll to Build\Debug.

2099.	[port]		win32: more manifiest issues.

	--- 9.3.3rc3 released ---

2096.	[bug]		libbind: handle applications that fail to detect
			res_init() failures better.

2095.	[port]		libbind: alway prototype inet_cidr_ntop_ipv6() and
			net_cidr_ntop_ipv6(). [RT #16388]
 
2094.	[contrib]	Update named-bootconf.  [RT# 16404]

2092.	[bug]		win32: dig, host, nslookup.  Use registry config
			if resolv.conf does not exist or no nameservers
			listed. [RT #15877] 

2091.	[port]		dighost.c: race condition on cleanup. [RT #16417]

2090.	[port]		win32: Visual C++ 2005 command line manifest support.
			[RT #16417]

2089.	[security]	Raise the minimum safe OpenSSL versions to
			OpenSSL 0.9.7l and OpenSSL 0.9.8d.  Versions
			prior to these have known security flaws which
			are (potentially) exploitable in named. [RT #16391]

2088.	[security]	Change the default RSA exponent from 3 to 65537.
			[RT #16391]

2086.	[port]		libbind: FreeBSD now has get*by*_r() functions.
			[RT #16403]

2085.	[doc]		win32: added index.html and README to zip. [RT #16201]

2084.	[contrib]	dbus update for 9.3.3rc2.

2083.	[port]		win32: Visual C++ 2005 support.

2082.	[doc]		Document 'cache-file' as a test only option.

	--- 9.3.3rc2 released ---

2081.	[port]		libbind: minor 64-bit portability fix in memcluster.c.
			[RT #16360]

2080.	[port]		libbind: res_init.c did not compile on older versions
			of Solaris. [RT #16363]

2076.	[bug]		Several files were missing #include <config.h>
			causing build failures on OSF. [RT #16341]

2074.	[bug]		dns_request_createvia2(), dns_request_createvia3(),
			dns_request_createraw2() and dns_request_createraw3()
			failed to send multiple UDP requests. [RT #16349]

2066.	[security]	Handle SIG queries gracefully. [RT #16300]

	--- 9.3.3rc1 released ---

2071.	[port]		Test whether gcc accepts -fno-strict-aliasing.
			[RT #16324]

2070.	[bug]		The remote address was not always displayed when
			reporting dispatch failures. [RT #16315]

2069.	[bug]		Cross compiling was not working. [RT #16330]

2067.	[bug]		'rndc' could close the socket too early triggering
			a INSIST under Windows. [RT #16317]

2065.	[bug]		libbind: probe for HPUX prototypes for
			endprotoent_r() and endservent_r().  [RT 16313]

2064.	[bug]		libbind: silence AIX compiler warnings. [RT #16218]

2063.	[bug]		Change #1955 introduced a bug which caused the first
			'rndc flush' call to not free memory. [RT #16244]

2062.	[bug]		'dig +nssearch' was reusing a buffer before it had
			been returned by the socket code. [RT #16307]

2057.	[bug]		Make setting "ra" dependent on both allow-query and
			allow-recursion. [RT #16290]

2056.	[bug]		dig: ixfr= was not being treated case insensitively
			at all times. [RT #15955]

2055.	[bug]		Missing goto after dropping multicast query.
			[RT #15944]

2054.	[port]		freebsd: do not explicitly link against -lpthread.
			[RT #16170]

2053.	[port]		netbsd:libbind: silence compiler warnings. [RT #16220]

2052.	[bug]		'rndc' improve connect failed message to report
			the failing address. [RT #15978]

2051.	[port]		More strtol() fixes. [RT #16249]

2050.	[bug]		Parsing of NSAP records was not case insensitive.
			[RT #16287]

2049.	[bug]		Restore SOA before AXFR when falling back from
			a attempted IXFR when transfering in a zone.
			Allow a initial SOA query before attempting
			a AXFR to be requested. [RT #16156]

2048.	[bug]		It was possible to loop forever when using
			avoid-v4-udp-ports / avoid-v6-udp-ports when
			the OS always returned the same local port.
			[RT #16182]

2047.	[bug]		Failed to initialise the interface flags to zero.
			[RT #16245]

2043.	[port]		nsupdate/nslookup: Force the flushing of the prompt
			for interactive sessions. [RT#16148]

2038.	[bug]		dig/nslookup/host was unlinking from wrong list
			when handling errors. [RT #16122]

2037.	[func]		When unlinking the first or last element in a list
			check that the list head points to the element to
			be unlinked. [RT #15959]

2036.	[bug]		'rndc recursing' could cause trigger a REQUIRE.
			[RT #16075]

2034.	[bug]		gcc: set -fno-strict-aliasing. [RT #16124]

	--- 9.3.3b1 released ---

2031.	[bug]		Emit a error message when "rndc refresh" is called on
			a non slave/stub zone. [RT # 16073]

2030.	[bug]		We were being overly conservative when disabling
			openssl engine support. [RT #16030]

2029.	[bug]		host printed out the server multiple times when
			specified on the command line. [RT #15992]

2028.	[port]		linux: socket.c compatability for old systems.
			[RT #16015]

2027.	[port]		libbind: Solaris x86 support. [RT #16020]

2026.	[bug]		Rate limit the two recursive client exceeded messages.
			[RT #16044]

2024.	[bug]		named emited spurious "zone serial unchanged"
			messages on reload. [RT #16027]

2023.	[bug]		"make install" should create ${localstatedir}/run and
			${sysconfdir} if they do not exist. [RT #16033]


contrib/query-loc-0.4.0/query-loc.c
#include        "loc.h"

/* $Id: query-loc.c,v 1.1.4.1 2008-02-15 02:19:42 marka Exp $ */

/* Global variables */
char *progname;
short debug;

int
main (argc, argv)
     int argc;
     char *argv[];
{
  extern char *optarg;
  extern int optind;

  short verbose = FALSE;
  char *host;

  char ch;

  char *loc = NULL;
  struct in_addr addr;
  struct hostent *hp;

  progname = argv[0];
  while ((ch = getopt (argc, argv, "vd:")) != EOF)
    {
      switch (ch)
	{
	case 'v':
	  verbose = TRUE;
	  break;
	case 'd':
	  debug = atoi (optarg);
	  if (debug <= 0)
	    {
	      (void) fprintf (stderr,
			      "%s: illegal debug value.\n", progname);
	      exit (2);
	    }
	  break;
	default:
	  usage ();
	}
    }
  argc -= optind;
  argv += optind;
  if (argc != 1)
    {
      usage ();
    }
  if (verbose || debug)
    {
      printf ("\nThis is %s, version %s.\n\n", progname, VERSION);
    }
  host = argv[0];
  (void) res_init ();

  if ((addr.s_addr = inet_addr (host)) == INADDR_NONE)
    {
      if (debug >= 1)
	printf ("%s is a name\n", host);
      loc = getlocbyname (host, FALSE);
    }
  else
    {
      if (debug >= 1)
	printf ("%s is an IP address ", host);
      hp = (struct hostent *) gethostbyaddr
	((char *) &addr, sizeof (addr), AF_INET);
      if (hp)
	{
	  if (debug >= 1)
	    printf ("and %s is its official name\n",
		    hp->h_name);
	  loc = getlocbyname (hp->h_name, FALSE);
	}
      else
	{
	  if (debug >= 1)
	    printf ("which has no name\n");
	  loc = getlocbyaddr (addr, NULL);
	}
    }
  if (loc == NULL)
    {
      printf ("No LOCation found for %s\n", host);
      exit (1);
    }
  else
    {
      if (verbose || debug)
	printf ("LOCation for %s is ", host);
      printf ("%s\n", loc);
      exit (0);
    }
}
